experiment/theorytest.json

```diff
@@ -0,0 +1,341 @@
+        },
+        {
+            "question": "14. In recursive algorithms, what is meant by 'tail recursion'?",
+            "answers": {
+                "a": "A type of recursion that is slower than regular recursion",
+                "b": "A recursive call that appears at the end of a function and doesn't require any further computation after the recursive call returns",
+                "c": "A recursive call that occurs within a loop",
+                "d": "A type of recursion that uses a tail data structure"
+            },
+            "correctAnswer": "b",
+            "explanations": {
+                "a": "This option is incorrect. Tail recursion is not necessarily slower than regular recursion; in fact, it can be more efficient in some cases.",
+                "b": "This option is correct. Tail recursion refers to a recursive call that appears at the end of a function and doesn't require any further computation after the recursive call returns. It is an important concept for optimization in some programming languages.",
+                "c": "This option is not the definition of tail recursion. A recursive call within a loop is a different concept and is not referred to as tail recursion.",
+                "d": "This option is unrelated to tail recursion. Tail recursion is about the position of the recursive call within the function's execution, not the use of a tail data structure."
+            },
+            "difficulty": "advanced"
+        },
+        {
+            "question": "15. How can you optimize a recursive function to reduce the overhead of function calls?",
+            "answers": {
+                "a": "By making the recursive calls non-tail recursive",
+                "b": "By using global variables instead of local variables",
+                "c": "By using memoization or dynamic programming techniques",
+                "d": "By removing the base case"
+            },
+            "correctAnswer": "c",
+            "explanations": {
+                "a": "This option is not a recommended way to optimize a recursive function for reducing function call overhead. Non-tail recursion can make optimization more challenging.",
+                "b": "This option is generally not recommended as using global variables can lead to unintended side effects and make the code less modular and maintainable.",
+                "c": "This option is correct. One of the primary ways to optimize a recursive function and reduce the overhead of function calls is by using memoization or dynamic programming techniques to store and reuse previously computed results.",
+                "d": "This option is incorrect. Removing the base case would make the recursive function incorrect and result in infinite recursion rather than optimization."
+            },
+            "difficulty": "advanced"
+        },
+        {
+            "question": "16. What is the time complexity of a recursive function that makes n recursive calls?",
```

Comment:
This question doesn't seem to be correct. The time complexity will also depend on the time taken by each instance of the function

Reply:
Resolved